### PR TITLE
support cosine in diskann

### PR DESCRIPTION
**Pull request by inabao**
#99

### CODE REVIEW

File: extern/diskann/DiskANN/src/disk_utils.cpp [hunk 1]
```diff
@@ -1039,29 +1039,6 @@ void create_disk_layout(const T *data, uint32_t npts, uint32_t ndims, const std:
         std::ifstream reorder_data_reader;
 
         uint32_t npts_reorder_file = 0, ndims_reorder_file = 0;
-        if (reorder_data_file != std::string(""))
```
Comment (jiaweizone) [inline extern/diskann/DiskANN/src/disk_utils.cpp:1042]: Why remove reorder logic?

File: extern/diskann/DiskANN/src/in_mem_data_store.cpp [hunk 2]
```diff
@@ -258,6 +264,11 @@ template <typename data_t> void InMemDataStore<data_t>::get_vector(location_t lo
     {
         loc = _loc_to_memory_index[loc];
         memcpy(dest, _data + loc * this->_dim, this->_dim * sizeof(data_t));
+        if (_compute_norms)
+        for (int i = 0; i < this->_dim; ++i)
```
Comment (jiaweizone) [inline extern/diskann/DiskANN/src/in_mem_data_store.cpp]: indent, add `{}` to `if` clause

Comment (inabao) [reply extern/diskann/DiskANN/src/in_mem_data_store.cpp]: done

File: extern/diskann/DiskANN/src/disk_utils.cpp (hunk 1 above)
Comment (inabao) [reply extern/diskann/DiskANN/src/disk_utils.cpp:1042]: This is the original reorder logic of DiskANN, which we did not rely on.

**Review by wxyucs — APPROVED** [review]
lgtm

**Pull request merged**